scala/README.md

```diff
@@ -477,9 +534,9 @@ It is definitely recommended to read the full Twitter's "[Effective Scala]" guid
 
     Note: beware that overriding the abstract `def` with a concrete `val` also suffers from the problem above.
 
-0. Avoid `lazy val`. `lazy val` is *not* free, or even cheap. Use it only if you absolutely need laziness semantics for correctness, not for "optimization". The init of a `lazy val` is super-expensive due to monitor acquisition cost, while every access is expensive due to `volatile`. Worse, `lazy val` [may deadlock](http://axel22.github.io/2013/06/10/on-lazy-vals.html).
+0. Avoid `lazy val`. `lazy val` is *not* free, or even cheap. Use it only when you absolutely need laziness semantics for correctness, never for "optimization". The initialization of a `lazy val` is super-expensive due to monitor acquisition cost, while every access is expensive due to `volatile`. Worse, `lazy val` [may deadlock](http://axel22.github.io/2013/06/10/on-lazy-vals.html) even when there are not circular dependencies between them.
```

Comment:
where does "monitor acquisition cost" come from, exactly? I didn't know monitors are involved with lazy vals at all. From what I've read the performance impact comes from memory fencing to uphold semantics regarding when the underlying volatile var containing the value can be read - the lock around that is cheap! ((on intel x86..))

also "are not" -> "are no"